pkg/kratos/handlers.go

```diff
@@ -206,10 +207,56 @@ func (a *API) handleUpdateFlow(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	shouldRegenerateBackupCodes, err := a.shouldRegenerateBackupCodes(r.Context(), cookies)
+	if err != nil {
+		err = fmt.Errorf("error when checking backup codes: %v", err)
+
+		a.logger.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if shouldRegenerateBackupCodes {
+		a.lookupSecretsSettingsRedirect(w, flowId)
+		return
+	}
```

Comment:
question: why do we set the cookies before we apply this logic? If we get an error, we will return a 500 but still initialize a session, which feels a little counter intuitive for me.